diff --git a/linux/drivers/dahdi/dahdi-base.c b/linux/drivers/dahdi/dahdi-base.c
index 5ea508c..010e9cf 100644
--- a/linux/drivers/dahdi/dahdi-base.c
+++ b/linux/drivers/dahdi/dahdi-base.c
@@ -53,6 +53,11 @@
 #include <linux/ktime.h>
 #include <linux/slab.h>
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 9, 0)
+# define HAVE_UNLOCKED_IOCTL 1
+# define HAVE_COMPAT_IOCTL 1
+#endif
+
 #if defined(HAVE_UNLOCKED_IOCTL) && defined(CONFIG_BKL)
 #include <linux/smp_lock.h>
 #endif
diff --git a/linux/drivers/dahdi/dahdi_transcode.c b/linux/drivers/dahdi/dahdi_transcode.c
index a495dcf..88e0a5f 100644
--- a/linux/drivers/dahdi/dahdi_transcode.c
+++ b/linux/drivers/dahdi/dahdi_transcode.c
@@ -38,6 +38,11 @@
 
 #include <dahdi/kernel.h>
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 9, 0)
+# define HAVE_UNLOCKED_IOCTL 1
+# define HAVE_COMPAT_IOCTL 1
+#endif
+
 static int debug;
 /* The registration list contains transcoders in the order in which they were
  * registered. */
